### ISSUE TITLE
gRPC: fully support server <-> client, server <-> server. 

### ISSUE THREAD

**Issue by siddontang** [pingcap/tikv#1433]
Refer #1424 #815 

/cc @disksing

**Comment by overvenus** [comment]
Rust grpc depends on tokio stack which requires mio to be **0.6** at least. For now, tikv uses mio **0.5**, an error occurred when I compile tikv. 🙁

```
$ cargo build
error: failed to select a version for `miow` (required by `mio`):
all possible versions conflict with previously selected versions of `miow`
  version 0.1.2 in use by miow v0.1.2
  possible versions to select: 0.1.5, 0.1.4, 0.1.3
```

```diff
diff --git a/Cargo.toml b/Cargo.toml
index fa2c0e6..7f646dc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -75,6 +75,9 @@ git = "https://github.com/pingcap/rust-prometheus.git"
 default-features = false
 features = ["nightly", "push"]
 
+[dependencies.grpc]
+git = "https://github.com/pingcap/grpc-rust.git"
+
```

Any ideas? @BusyJay @siddontang 

CC #1430

**Comment by BusyJay** [comment]
You need to run `cargo update -p miow` first.

**Comment by overvenus** [comment]
@BusyJay  Thanks! It works like a charm, I thought it would not be solved until #1430.
